### ISSUE TITLE
Update and preserve user data

### ISSUE THREAD

**Issue by belt-drive** [dabreegster/abstreet#216]
This came up in a separate bug report recently. I want to update to the version of the program that I'm running while maintaining my user data. Putting in a report to make this a more streamlined task in the future. For now, should I copy out my user data folder and then move it back in after reinstalling?

**Comment by dabreegster** [comment]
Yep, try copying over `data/player/`. `camera_state` is not so interesting (it's just where you were focused the last time on a map). `edits` is more useful.

For now, you'll notice some of your edits silently don't show up in the loader. That's because they're no longer considered valid. Sometimes that's a bug like #113, or it could be that the underlying OSM data has changed in the new version, and your edit to the third lane from the center on some road no longer makes sense, because that road only has two lanes, it turns out.

There are at least a few things to do here:

1) Come up with a better story for grabbing `data/player/` from an older install. Might be tough to do automatically without installing to some canonical location, maybe it'll just be a file chooser dialog that you manually point at old data.

2) List all edits in the directory, no matter what. When you try to load a broken one, explain clearly what happened.

3) Try to automatically repair outdated edits. Probably the simplest thing is to skip over commands that no longer make sense. That might have confusing ripple effects sometimes; your changes to a traffic signal depend on lane edits done first.
